google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/CloudSdkAppEngineHelper.java

```diff
@@ -79,61 +139,148 @@ public File defaultDockerfile(DeploymentArtifactType deploymentArtifactType) {
+
+    return new CloudSdk.Builder()
+        .sdkPath(getGcloudCommandPath())
+        .async(true)
+        .addStdErrLineListener(stdErrListener)
+        .addStdOutLineListener(stdOutListener)
+        .exitListener(exitListener)
+        .startListener(startListener)
+        .appCommandCredentialFile(credentialsPath)
+        .appCommandMetricsEnvironment("gcloud-intellij")
+        .appCommandMetricsEnvironmentVersion(pluginInfoService.getPluginVersion())
+        .appCommandGsUtil(0)
+        .appCommandOutputFormat("json")
+        .build();
+  }
+
+  private static final String CLIENT_ID_LABEL = "client_id";
+  private static final String CLIENT_SECRET_LABEL = "client_secret";
+  private static final String REFRESH_TOKEN_LABEL = "refresh_token";
+  private static final String GCLOUD_USER_TYPE_LABEL = "type";
+  private static final String GCLOUD_USER_TYPE = "authorized_user";
+
+  @Override
+  public void stageCredentials(String googleUsername) {
+    CredentialedUser projectUser = Services.getLoginService().getAllUsers()
+        .get(googleUsername);
+
+    GoogleLoginState googleLoginState;
+    if (projectUser != null) {
+      googleLoginState = projectUser.getGoogleLoginState();
+    } else {
+      return;
+    }
+    String clientId = googleLoginState.fetchOAuth2ClientId();
+    String clientSecret = googleLoginState.fetchOAuth2ClientSecret();
+    String refreshToken = googleLoginState.fetchOAuth2RefreshToken();
+    Map<String, String> credentialMap = ImmutableMap.of(
+        CLIENT_ID_LABEL, clientId,
+        CLIENT_SECRET_LABEL, clientSecret,
+        REFRESH_TOKEN_LABEL, refreshToken,
+        GCLOUD_USER_TYPE_LABEL, GCLOUD_USER_TYPE
     );
+    String jsonCredential = new Gson().toJson(credentialMap);
+    try {
+      credentialsPath = FileUtil
+          .createTempFile(
+              "tmp_google_application_default_credential",
+              "json",
+              true /* deleteOnExit */);
+      Files.write(jsonCredential, credentialsPath, Charset.forName("UTF-8"));
+    } catch (IOException ex) {
+      throw new RuntimeException(ex);
+    }
+  }
+
+  @Override
+  public void deleteCredentials() {
+    if (credentialsPath != null && credentialsPath.exists()) {
+      if (!credentialsPath.delete()) {
+        logger.warn("failed to delete credential file expected at " + credentialsPath.getPath());
+      }
+    }
   }
 
   @NotNull
   private DeploymentOperationCallback wrapCallbackForUsageTracking(
       final DeploymentOperationCallback deploymentCallback,
-      ConfigType deploymentType, DeploymentArtifactType artifactType) {
+      AppEngineDeploymentConfiguration deploymentConfiguration,
+      File artifactToDeploy) {
 
-    StringBuilder labelBuilder = new StringBuilder("deploy.flex");
-    switch (deploymentType) {
-      case AUTO:
+    StringBuilder labelBuilder = new StringBuilder("deploy");
+
+    if (environment == Environment.APP_ENGINE_STANDARD) {
```

Comment:
I do respect the way Chanseok made his code fail if a new Environment value gets added. Although I do admit that possibility seems remote.